test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,460 @@
+        ), f"Couldn't find OpInfo for {decorate_meta}. Did you need to specify variant_name?"
+        decorators = list(opinfo.decorators)
+        new_decorator = opinfo_core.DecorateInfo(
+            decorate_meta.decorator,
+            test_case_name,
+            base_test_name,
+            device_type=decorate_meta.device_type,
+            dtypes=decorate_meta.dtypes,
+        )
+        decorators.append(new_decorator)
+        opinfo.decorators = tuple(decorators)
+
+    # This decorator doesn't modify fn in any way
+    def wrapped(fn):
+        return fn
+
+    return wrapped
+
+
+def get_torch_op_name(op: Union[str, Callable]) -> str:
+    """Returns the name of the torch function corresponding to the given op."""
+    if callable(op):
+        module_name = op.__module__.split("torch.", 1)
+        op_name = op.__name__
+        if len(module_name) == 2:
+            # Remove the torch. prefix
+            op_name = f"{module_name[1]}.{op_name}"
+        return op_name
+    # Already a string
+    return op
+
+
+def opsets_before(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is before the specified."""
+
+    def compare(other_opset: int):
+        return other_opset < opset
+
+    return compare
+
+
+def opsets_after(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is after the specified."""
+
+    def compare(other_opset: int):
+        return other_opset > opset
+
+    return compare
+
+
+def reason_onnx_runtime_does_not_support(
+    operator: str, dtypes: Optional[Sequence[str]] = None
+) -> str:
+    """Formats the reason: ONNX Runtime doesn't support the given dtypes."""
+    return f"{operator} on {dtypes or 'dtypes'} not supported by ONNX Runtime"
+
+
+def reason_onnx_does_not_support(
+    operator: str, dtypes: Optional[Sequence[str]] = None
+) -> str:
+    """Formats the reason: ONNX doesn't support the given dtypes."""
+    return f"{operator} on {dtypes or 'dtypes'} not supported by the ONNX Spec"
+
+
+def reason_jit_tracer_error(info: str) -> str:
+    """Formats the reason: JIT tracer errors."""
+    return f"JIT tracer error on {info}"
+
+
+def reason_flaky() -> str:
+    """Formats the reason: test is flaky."""
+    return "flaky test"
+
+
+# Modify this section ##########################################################
+# NOTE: Modify this section as more ops are supported. The list should be sorted
+# alphabetically.
+#
+# For example, to add a test for torch.ceil:
+# 1.  Add `torch.ceil` to ALLOWLIST_OP then run pytest.
+#         You can also add a string, e.g. "ceil" or "__radd__".
+# 2a. If the test fails, fix the error or add a new entry to EXPECTED_SKIPS_OR_FAILS.
+# 2b. If the test is expected to fail only on certain opsets, add a new entry to
+#     EXPECTED_OPSET_FAILS.
+
+# Ops to be tested for consistency between onnx and pytorch
+ALLOWLIST_OP: AbstractSet[str] = frozenset(
```

Comment:
Having an allowlist of operators or even particular OpInfos by name is totally fine, and many tests filter the OpInfos into their own subset of interest.